alacritty_terminal/src/term/mod.rs

```diff
@@ -205,21 +129,23 @@ pub struct RenderableCellsIter<'a, C> {
     config: &'a Config<C>,
     colors: &'a color::List,
     selection: Option<SelectionRange<Line>>,
+    search: RenderableSearch<'a>,
 }
 
 impl<'a, C> RenderableCellsIter<'a, C> {
     /// Create the renderable cells iterator.
     ///
     /// The cursor and terminal mode are required for properly displaying the
     /// cursor.
-    fn new<'b, T>(
-        term: &'b Term<T>,
-        config: &'b Config<C>,
+    fn new<T>(
+        term: &'a Term<T>,
+        config: &'a Config<C>,
         selection: Option<SelectionRange>,
-    ) -> RenderableCellsIter<'b, C> {
+    ) -> RenderableCellsIter<'a, C> {
         let grid = &term.grid;
 
-        let inner = grid.display_iter();
+        let viewport_end = grid.display_offset();
+        let viewport_start = viewport_end + grid.num_lines().0 - 1;
```

Comment:
Could you move those checks inside closure? there's no need to compute them without having selection.